lib/Conversion/BGVToOpenfhe/BGVToOpenfhe.cpp

```diff
@@ -0,0 +1,264 @@
+#include "include/Conversion/BGVToOpenfhe/BGVToOpenfhe.h"
+
+#include "include/Dialect/BGV/IR/BGVDialect.h"
+#include "include/Dialect/BGV/IR/BGVOps.h"
+#include "include/Dialect/BGV/IR/BGVTypes.h"
+#include "include/Dialect/LWE/IR/LWEAttributes.h"
+#include "include/Dialect/LWE/IR/LWEDialect.h"
+#include "include/Dialect/LWE/IR/LWEOps.h"
+#include "include/Dialect/LWE/IR/LWETypes.h"
+#include "include/Dialect/Openfhe/IR/OpenfheDialect.h"
+#include "include/Dialect/Openfhe/IR/OpenfheOps.h"
+#include "include/Dialect/Openfhe/IR/OpenfheTypes.h"
+#include "lib/Conversion/Utils.h"
+#include "mlir/include/mlir/Dialect/Arith/IR/Arith.h"   // from @llvm-project
+#include "mlir/include/mlir/Dialect/Func/IR/FuncOps.h"  // from @llvm-project
+#include "mlir/include/mlir/Transforms/DialectConversion.h"  // from @llvm-project
+
+namespace mlir::heir::bgv {
+
+#define GEN_PASS_DEF_BGVTOOPENFHE
+#include "include/Conversion/BGVToOpenfhe/BGVToOpenfhe.h.inc"
+
+class ToLWECiphertextTypeConverter : public TypeConverter {
+ public:
+  // Convert ciphertext to RLWE ciphertext
+  ToLWECiphertextTypeConverter(MLIRContext *ctx) {
+    addConversion([](Type type) { return type; });
+    addConversion([ctx](CiphertextType type) -> Type {
+      assert(type.getLevel().has_value());
+      auto level = type.getLevel().value();
+      assert(level < type.getRings().getRings().size());
+
+      auto ring = type.getRings().getRings()[level];
+      auto cmod = ring.getCmod();
+      auto polyDegree = ring.getIdeal().getDegree();
+      auto dim = type.getDim();
+      return lwe::RLWECiphertextType::get(
+          // FIXME: What should be the default encoding?
```

Comment:
Let's just change this to a TODO(#99) to set a default encoding and we can submit the rest of this PR as-is.